### PR TITLE
= agent-test: add some integration tests + minor bugfix for mixin generation

### PR DESCRIPTION
**Pull request by cspinetta**

### CODE REVIEW

File: agent-test/build.gradle [hunk 1]
```diff
@@ -32,6 +32,8 @@ dependencies {
 
     compile 'io.kamon:kanela-agent-attacher:0.0.10'
     compile 'io.kamon:kanela-agent:0.0.10'
+    compile 'io.kamon:kanela-scala-extension_2.12:0.0.10'
+    compile 'io.kamon:kamon-autoweave_2.12:0.6.5'
```
Comment (dpsoft) [inline agent-test/build.gradle]: this dependency is not necessary, with `kanela-agent-attacher` is enough

File: agent-test/src/test/scala/app/kanela/specs/AttachInRuntimeSpec.scala [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+import scala.collection.mutable.ListBuffer
+
+class AttachInRuntimeSpec extends FlatSpec with Matchers with BeforeAndAfterAll {
+
+  "Kamon agent" should "be able to attach in runtime and instrument the loaded classes" in {
+    val testClass = new TestClass()
+    testClass.addValue(ListBuffer()) shouldBe ListBuffer("body")
+
+    // attach agent
+    AgentLoader.attachAgentToJVM(classOf[Kanela])
```
Comment (dpsoft) [inline agent-test/src/test/scala/app/kanela/specs/AttachInRuntimeSpec.scala]: with `kanela-attacher` you can do `Kanela.attach()` ;)

File: agent-test/src/test/scala/app/kanela/specs/AgentLoader.scala [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+
+import kamon.autoweave.loader.AttachmentProviders
+
+import scala.util.control.NoStackTrace
+
+
+/**
+  * Temporal implementation, until kamon-autowave is fixed.
+  */
+object AgentLoader {
```
Comment (dpsoft) [inline agent-test/src/test/scala/app/kanela/specs/AgentLoader.scala]: It's not necessary

**Comment by cspinetta** [comment]
@dpsoft please, not merge. It's failing some tests. e.g.:

```bash
./gradlew :agent-test:simple-instrumentation-spec
```

Result in:

```bash
[ScalaTest-2-running-SimpleInstrumentationSpec] INFO 2018-04-01 13:46:02  Logger : Transformed => class app.kanela.cases.simple.TestClass and loaded from sun.misc.Launcher$AppClassLoader 
*** RUN ABORTED *** (631 milliseconds)
  ...
  Cause: java.lang.ClassFormatError: StackMapTable format error: bad verification type
  at app.kanela.specs.SimpleInstrumentationSpec.$anonfun$new$1(SimpleInstrumentationSpec.scala:27)
  at org.scalatest.OutcomeOf.outcomeOf(OutcomeOf.scala:85)
  ...

```

**Pull request merged**